sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/BufferedHttpResponse.java

```diff
@@ -50,7 +50,7 @@ public HttpHeaders headers() {
 
     @Override
     public Flux<ByteBuf> body() {
-        return bodyAsByteArray().flatMapMany(bytes -> Flux.just(Unpooled.wrappedBuffer(bytes)));
+        return bodyAsByteArray().flatMapMany(bytes -> Flux.just(Unpooled.wrappedBuffer(bytes).retain()));
```

Comment:
When searching for Unpooled.wrappedBuffer there are several other places where we don't retain it as well.  SHouldn't this be consistent across the board?

Reply:
I assume you saw this one? https://github.com/netty/netty/issues/1482

Reply:
I assume calling retain here assuming the factory methods to create the ByteBuf are only called when write etc. is planned later is acceptable - but teh pattern should be consistent - or if there are reaosns to not be consistent everywhere we need to add comments or refactor the code so that the intentions become clear

Reply:
This is done, resolving this conversation.